### ISSUE TITLE
OS X, FAIL: 13-verify-perfect-nolut-float.test

### ISSUE THREAD

**Issue by novelistparty** [kamalmostafa/minimodem#24]
I compiled from source, minimodem-0.23, downloaded from http://www.whence.com/minimodem/  
    `./configure --disable-dependency-tracking --without-alsa --without-pulseaudio`

tests/test-suite.log:

```
==========================================
   minimodem 0.23: tests/test-suite.log
==========================================

# TOTAL: 26
# PASS:  25
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: 13-verify-perfect-nolut-float
===================================

IMPERFECT ndata=500 confidence=17874648.000 ampl=1.000 bps=1200.00 (rate perfect)
FAIL 13-verify-perfect-nolut-float.test (exit status: 1)
```

**Comment by kamalmostafa** [comment]
Thanks for reporting this, John.  I'm not sure about the cause of the test case fail (13-verify-perfect-nolut-float) you're getting.  I don't have an OS X platform to work on it, and it doesn't manifest on other platforms.

However, I did identify and fix a problem that may be related (for reference that fix is commit ea3f837c90afaeb3334514f2f33fe677998adab9).  Please try the just-released minimodem-0.24 (which includes that fix and new test cases for it).  If all test cases pass now, great.  If not, please send me all of the resulting `tests/*.log` files.

**Comment by kebno** [comment]
From `test-suite.log` for minimodem-0.24 (archive of all .log files attached below):

```
FAIL: 13-verify-perfect-nolut-float
===================================

IMPERFECT ndata=500 confidence=17874648.000 ampl=1.000 bps=1200.00 (rate perfect)
FAIL 13-verify-perfect-nolut-float.test (exit status: 1)

FAIL: 16-verify-tx-consistent
=============================

./16-verify-tx-consistent.test: line 22: md5sum: command not found
FAIL 16-verify-tx-consistent.test (exit status: 127)

FAIL: 17-verify-tx-consistent-float
===================================

/Users/john/src/minimodem-0.24/tests/16-verify-tx-consistent.test: line 22: md5sum: command not found
FAIL 17-verify-tx-consistent-float.test (exit status: 127)
```

[minimodem-0.24_test_logs.tar.gz](https://github.com/kamalmostafa/minimodem/files/233123/minimodem-0.24_test_logs.tar.gz)

**Comment by kebno** [comment]
Ha, I just read the output. 
On OSX, there is no `md5sum` it is just [`md5`](https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man1/md5.1.html)

**Comment by kamalmostafa** [comment]
Ok, so minimodem-0.24 does not fix test case 13 for OS X (the fix in commit ea3f837 wasn't related after all).  Still an open issue.

While it certainly won't fix the test-case 13 problem, new commit d4ce821 should allow you to run test cases 16 and 17 (and I'd still like to know if they pass or fail on OS X).  John, can you apply it and test?

**Comment by kebno** [comment]
With commit d4ce821, test cases 16 and 17 pass. 
13 does not pass.
Attached is the test log.
[minimodem-d4ce821_test_logs.tar.gz](https://github.com/kamalmostafa/minimodem/files/233617/minimodem-d4ce821_test_logs.tar.gz)

**Comment by rgov** [comment]
On macOS Mojave 10.14.6, I tried:

```
$ autoreconf -I
$ ./configure --disable-dependency-tracking --without-alsa --without-pulseaudio
$ make
$ make check
============================================================================
Testsuite summary for minimodem 0.24
============================================================================
# TOTAL: 28
# PASS:  28
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
```

It looks like this could be closed now?

**Comment by kamalmostafa** [comment]
Thanks for the report and confirmation folks.  I'll consider this as having been fixed by 0.24 unless someone says otherwise.
